File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void){
    
    float change;
    do{
        printf("O hai! How much change is owed?\n");
        change = get_float();
    }
    while(change < 0);
    
    int intChange = lroundf (100*change);
    int counter = 0;
    int i = 0;
    while (intChange >= 25){
        i+= (intChange - intChange % 25)/25;
        intChange-= 25*i;
        counter+=i;
        break;
    }
    i = 0;
    while (intChange >= 10){
        i+= (intChange - intChange % 10)/10;
        intChange-= 10*i;
        counter+=i;
        break;
    }
    i = 0;
    while (intChange >= 5){
        i+= (intChange - intChange % 5)/5;
        intChange-= 5*i;
        counter+=i;
        break;
    }
    i = 0;
    while (intChange >= 1){
        counter+=intChange;
        break;
    }
        
    printf("%i\n", counter);
}

File: pset6/greedy.py
# Calculates the minimum number of coints required to give a user change

import cs50

def main():
    change = -1
    while change < 0:
        print("O hai! How much change is owed?")
        change = cs50.get_float()
    
    intChange = 100*change
    counter = 0
    i = 0
    coins = [25, 10, 5, 1]
    for coin in coins:
        while intChange >= coin:
            i+=(intChange - intChange % coin)/coin
            intChange-= coin*i
            counter+=i
            break
        i = 0
    print("{0:.0f}".format(counter))

if __name__ == '__main__':
    main()